cpp/src/parquet/encoding.cc

```diff
@@ -3382,6 +3385,20 @@ class DeltaByteArrayDecoderImpl : public DecoderImpl, virtual public TypedDecode
       if (ARROW_PREDICT_FALSE(static_cast<size_t>(prefix_len_ptr[i]) > prefix.length())) {
         throw ParquetException("prefix length too large in DELTA_BYTE_ARRAY");
       }
+      if (prefix_len_ptr[i] == 0) {
+        prefix = std::string_view{buffer[i]};
+        continue;
+      }
+      if (buffer[i].len == 0 && prefix.data() != last_value_.data()) {
```

Comment:
`prefix.data() == last_value_.data()` is only for the first loop iteration, right?
If so, how about we move the first iteration before the loop? This would probably simplify the loop itself, and may make performance better as well.